src/RevProc.cc

```diff
@@ -16,18 +16,19 @@ using MemSegment = RevMem::MemSegment;
 
 RevProc::RevProc( unsigned Id,
                   RevOpts *Opts,
+                  unsigned NumHarts,
                   RevMem *Mem,
                   RevLoader *Loader,
-                  std::vector<std::shared_ptr<RevThread>>& AssignedThreads,
+                  std::unordered_map<uint32_t, std::shared_ptr<RevThread>>& AssignedThreads,
                   std::function<uint32_t()> GetNewTID,
                   RevCoProc* CoProc,
                   SST::Output *Output )
   : Halted(false), Stalled(false), SingleStep(false),
     CrackFault(false), ALUFault(false), fault_width(0),
-    id(Id), HartToDecode(0), HartToExec(0), Retired(0x00ull),
-    opts(Opts), mem(Mem), loader(Loader), AssignedThreads(AssignedThreads),
-    GetNewThreadID(GetNewTID), output(Output), feature(nullptr),
-    PExec(nullptr), sfetch(nullptr) {
+    id(Id), HartToDecode(0), HartToExec(0), Retired(0x00ull), 
+    numHarts(NumHarts), opts(Opts), mem(Mem), loader(Loader), 
+    AssignedThreads(AssignedThreads), GetNewThreadID(GetNewTID), 
+    output(Output), feature(nullptr), PExec(nullptr), sfetch(nullptr) {
```

Comment:
By the way, you can use constructor parameter names the same as member names.
```c++
: numHarts(numHarts), opts(opts), mem(mem), loader(loader), 
AssignedThreads(AssignedThreads), GetNewThreadID(GetNewTID), 
output(output), feature(nullptr), PExec(nullptr), sfetch(nullptr) {
```
There is no need to distinguish the constructor parameter names from the member names. In the initialization list (the `: var1(value1), var2(value2), ...`), the member names are in the class scope, while their initialization expressions are in the constructor function scope, so if they are the same, there is no conflict.

Reply:
Ah okay. Well that's good to know.

Reply:
I only updated my `numHarts` one but a future PR can feel free to clean that up